pkg/scheduler/core/generic_scheduler.go

```diff
@@ -286,21 +280,22 @@ func (g *genericScheduler) Preempt(ctx context.Context, prof *profile.Profile, s
 		klog.Infof("%v potential nodes for preemption, first %v are: %v", len(potentialNodes), len(sample), sample)
 	}
 	var pdbs []*policy.PodDisruptionBudget
-	if g.pdbLister != nil {
-		pdbs, err = g.pdbLister.List(labels.Everything())
+	pdbLister := fh.SharedInformerFactory().Policy().V1beta1().PodDisruptionBudgets().Lister()
```

Comment:
Do not assume PDBs are available. The API might not be enabled on a GA-only cluster

Reply:
Make sure the feature is enabled (maybe like GetPodDisruptionBudgetLister did?) before accessing the lister like this, otherwise the scheduler will list a non-existent PDB API every second and get a 404 error forever.

Reply:
Ah, thanks. I did the featuregate check in a followup PR, but not this one. Let me update.

Reply:
This explains why the followup (#92049) passed CI, but not this one.

Reply:
GetPodDisruptionBudgetLister cannot be used in pkg/scheduler/core due to cycling dependency, so I put featuregate check here.